### ISSUE TITLE
`matchesSnapshot`

### ISSUE THREAD

**Issue by tomasbjerre** [assertj/assertj#2471]
#### Summary
I am missing a way of asserting a big string of information. I like how Jest has implemented [snapshot testing](https://jestjs.io/docs/snapshot-testing). And in Java I usually use [ApprovalTests.Java](https://github.com/approvals/ApprovalTests.Java).

Would be nice if this feature can be part of AssertJ so that I don't need an extra library.

#### Example

Perhaps I capture the body being sent to an integration, and I want to assert that whole body.

```java
package se.bjurr.project
...
public class MyTestClass {
    public void myTest() {
        assertThat(actualBigString)
            .matchesSnapshot()
    }
}
```

Here I want  a file to be created `src/main/java/se/bjurr/project/MyTestClass.myTest.approved.txt`. I can verify its correctness when committing it to git.

I prefer these tests to never fail, just create the approved file and let me decide if it is ok when I commit it. But many people will probably want to have a file created like `src/main/java/se/bjurr/project/MyTestClass.myTest.received.txt` and only have `src/main/java/se/bjurr/project/MyTestClass.myTest.approved.txt` created if there is no approved file.

**Comment by scordio** [comment]
This doesn't seem to fit AssertJ's main purpose. What problem do you see with adding a library created for this?

It seems a similar topic to assertions with JUnit - basic support is provided, but if you want more you should use [a third-party library](https://junit.org/junit5/docs/current/user-guide/#writing-tests-assertions-third-party).

**Comment by tomasbjerre** [comment]
I started fiddling with this, implementing it as an extension to AssertJ.

Code is here:
https://github.com/tomasbjerre/assertj-snapshot

Is there any chance of this repo being moved to this organization?

**Comment by scordio** [comment]
I still feel this doesn't belong to AssertJ, considering that solutions like [`java-snapshot-testing`](https://github.com/origin-energy/java-snapshot-testing) exist already.

Of course, nothing prevents having an AssertJ flavor of snapshot testing, similarly to what [JSonUnit does](https://github.com/lukas-krecan/JsonUnit#assertj). Have you thought about asking `java-snapshot-testing` for it?

**Comment by vlsi** [comment]
+1 to what @tomasbjerre says.

Basically, `assertThat(...).isEqualTo(...)` is not really different from `assertThat(...).snapshotMatches(...)`.

They both compare expected value with actual value.
They both aim for displaying human-readable output.
They both aim for having `<click to see difference>` link in IDEs.
They both aim for having customizable exclude/include behaviours (e.g. ignore certain parameters or ignore newline style).

For instance, a really nice UI feature approvaltests-java has is it automatically opens a diff window in UI: https://github.com/approvals/ApprovalTests.Java/blob/18.5.0/approvaltests/docs/Reporters.md#supported-diff-tools

In other words, you launch a test, and it automatically opens a diff view should the test fail.
I believe that would be nice to have in AssertJ as well.
